src/main/java/com/rtb/projectmanagementtool/user/UserServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.rtb.projectmanagementtool.user;
+
+import com.google.appengine.api.datastore.*;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Test user
+    // long userID1 = 1l;
+    // long AuthID1 = 1l;
+    // String userName1 = "Jake";
+    // long userYear1 = 2023;
+    // ArrayList<String> userMajors1 = new ArrayList<>(Arrays.asList("Geology", "Physics"));
+    // long userTotalCompTasks1 = 8;
+    // UserData user1 =
+    // new UserData(userID1, AuthID1, userName1, userYear1, userMajors1, userTotalCompTasks1);
+
+    // get the User
+    long userID = Long.parseLong(request.getParameter("userID"));
+    UserController userController = new UserController(datastore);
+    UserData user = UserController.getUserByID(userID);
```

Comment:
"non-static method getUserByID(long) cannot be referenced from a static context": I think I understand the difference between static/non static from some research I did this morning but I'm still a bit confused. getUserByID is defined in UserController and is only called in the server AFTER the UserController has been declared. Isn't this technically the structure that would allow for this call?

Reply:
` UserData user = UserController.getUserByID(userID);`  the **U** in UserController is capital, so it would reference the class in a static context and not your actual local instance with the lower case **u**.